src/contracts/mixins/GPv2Signing.sol

```diff
@@ -143,7 +143,8 @@ abstract contract GPv2Signing {
             owner = recoverEthsignSigner(orderDigest, signature);
         } else if (signingScheme == Scheme.Eip1271) {
             owner = recoverEip1271Signer(orderDigest, signature);
-        } else if (signingScheme == Scheme.PreSign) {
+        } else {
+            // signingScheme == Scheme.PreSign
             owner = recoverPreSigner(orderDigest, signature, order.validTo);
         }
```

Comment:
I'm not a huge fan of this change as it is semantically different than what it was previously. 

Previously, if we added a signing scheme but didn't implement a separate branch in this `elif`, then it would recover owner address 0. With this change this is no longer the case, and it would try to recover with a pre-signature. To me, this is a limitation of the coverage report and would slighty prefer a comment explaining why we don't have 100% coverage for this branch and how its impossible to get there (because solidity panics when reading an enum with an invalid value). To illustrate, the following always reverts when reading `value`:

```solidity
contract Foo {
    enum MyEnum { Bar, Baz }
    function doStuff(MyEnum value) external pure returns (bool check) {
        assembly {
            value := 42
        }
        if (value == MyEnum.Bar) {
            check = true;
        } else if (value == MyEnum.Baz) {
            check = true;
        }
    }
}
```

Reply:
I see your point, but I'd argue that the changes are right for the current code and the coverage tool is correct at reporting an unreachable branch.
I agree that this is not great in light of future changes. I'd say this is more a limitation of our test suite: I wanted to add a test checking that the enum test fails for `value = 4`, but this failure cannot be distinguished from some other kind of failure (e.g., bad signature). Returning the zero address on failure is also a hack relying on the expectation that the zero address cannot really be part of a trade: it would revert when trying to withdraw.

I personally am ok with assuming that we are going to remember to add new tests if we add a new signature scheme. Assuming this does not happen, I prefer that the behavior falls back to an existing signature scheme (meaning that the behavior should have been already integrated with the code assumptions) rather than saying that the signature verifies to the zero address.

Reply:
If we want to make this code strictly future proof, I'd be ok with a change like:
```
        } else if (signingScheme == Scheme.PreSign) {
            owner = recoverPreSigner(orderDigest, signature, order.validTo);
        } else {
            // code coverage explanation
            revert("GPv2: invalid signing scheme.");
        }
```

Reply:
Those are good points. Lets keep as is right now with the `else`.

Reply:
Should we maybe assert in the branch that `signingScheme == Scheme.PreSign`? This way it would fail if we added a new scheme and ran at least one test with it. I also like the suggestion of adding a specific always reverting else branch (assuming there is no gas overhead).

Reply:
> I also like the suggestion of adding a specific always reverting else branch 

The issue there is that the coverage tool will always determine this as an "untested branch".

> (assuming there is no gas overhead).

I think there might be a slight deployment overhead (a tiny bit of extra code), but I don't suspect there will be runtime overhead, unless of course it affects decisions made by the optimizer.